pkg/scheduler/uid/uid.go

```diff
@@ -15,55 +15,46 @@ type UID struct {
 	ser   string
 }
 
-func New(group uint64, name string) *UID {
-	if name == "" {
-		name = uuid.New()
-	}
-	return &UID{
+func New(group uint64, name string) UID {
+	return UID{
 		group: group,
 		name:  name,
 		ser:   fmt.Sprintf("%x_%s", group, name),
 	}
 }
 
-func (self *UID) Name() string {
-	if self != nil {
-		return self.name
-	}
-	return ""
+func (u UID) Name() string {
+	return u.name
 }
 
-func (self *UID) Group() uint64 {
-	if self != nil {
-		return self.group
-	}
-	return 0
+func (u UID) Group() uint64 {
+	return u.group
 }
 
-func (self *UID) String() string {
-	if self != nil {
-		return self.ser
-	}
-	return ""
+func (u UID) String() string {
+	return u.ser
 }
 
-func Parse(ser string) *UID {
+func Parse(ser string) (UID, error) {
 	parts := strings.SplitN(ser, "_", 2)
 	if len(parts) != 2 {
-		return nil
+		return UID{}, errors.New("invalid UID format (expected <uint64>_<string>)")
 	}
 	group, err := strconv.ParseUint(parts[0], 16, 64)
 	if err != nil {
-		log.Errorf("illegal UID group %q: %v", parts[0], err)
-		return nil
+		return UID{}, fmt.Errorf("invalid UID group %q: %v", parts[0], err)
 	}
-	if parts[1] == "" {
-		log.Errorf("missing UID name: %q", ser)
-		return nil
+	name := parts[1]
+	if name == "" {
+		return UID{}, fmt.Errorf("invalid UID name %q (expected non-empty)", name)
```

Comment:
It's logging the whole id, not just the name. Tho I see how the colon following "name" might be confusing.